### ISSUE TITLE
PostCSS could generate wrong source map error

### ISSUE THREAD

**Issue by mburger81** [ionic-team/ionic#13763]
**Ionic version:**  (check one with "x")
(For Ionic 1.x issues, please use https://github.com/ionic-team/ionic-v1)
[ ] **2.x**
[x] **3.x**
[ ] **4.x**

**I'm submitting a ...**  (check one with "x")
[x] bug report
[ ] feature request

Please do not submit support requests or "How to" questions here. Instead, please use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

**Current behavior:**
If we build a new app using your blank starter or any other template building the android app we are running into this error

```
[16:52:18]  sass started ... 
Witout `from` option PostCSS could generate wrong source map or do not find Browserslist config. Set it to CSS file path or to `undefined` to prevent this warning
[16:52:19]  sass finished in 1.13 s 
```

**Expected behavior:**
The bug message should not be

**Steps to reproduce:**
Create a new blank app using
`ionic start postcss`
build the android app doing
`ionic cordova build android`

and you can see log with the warn or error message
```
[16:57:54]  sass started ... 
Witout `from` option PostCSS could generate wrong source map or do not find Browserslist config. Set it to CSS file path or to `undefined` to prevent this warning
[16:57:55]  sass finished in 793 ms 
```

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):

```
cli packages: (/usr/lib/node_modules)

    @ionic/cli-utils  : 1.19.0
    ionic (Ionic CLI) : 3.19.0

global packages:

    cordova (Cordova CLI) : 8.0.0 

local packages:

    @ionic/app-scripts : 3.1.6
    Cordova Platforms  : android 7.0.0
    Ionic Framework    : ionic-angular 3.9.2

System:

    Android SDK Tools : 26.1.1
    Node              : v8.9.4
    npm               : 5.6.0 
    OS                : Linux 4.13

Environment Variables:
    ANDROID_HOME : /abcdef/android/sdk

Misc:
    backend : legacy
```

**Comment by StefanRein** [comment]
As the warning says:

> Without `from` option PostCSS could generate wrong source map or do not find Browserslist config. **Set it to CSS file path or to `undefined` to prevent this warning**

I added the field `from` and did set it to undefined and created a pull request.

https://github.com/StefanRein/ionic-app-scripts/blob/29873ae96611ae209ca4b065fd9940d6b7b96bda/src/sass.ts#L285

This resolves this warning. But I don't know if this has further consequences and the Ionic Team needs to fill in there a path to the generated css file or something like that.

@danbucholtz Could you review this please? Our automated builds are failing because of this. 
(The "warning" is interpreted as an error..)

**Comment by StefanRein** [comment]
Until this is fixed, 
run the command for reverting back some patches before the error was introduced:

`npm i -D -E postcss@6.0.14` (save exact this version before the warning)

[CHANGELOG.md of PostCSS](https://github.com/postcss/postcss/blob/master/CHANGELOG.md)

> **6.0.16**
> Fix warning text (by Michael Keller).
> **6.0.15**
> Add warning about missed from option on process().then() call.
> Add IE 10 support.

https://github.com/postcss/postcss/blame/master/lib/lazy-result.es6#L192

In the: [package.json of ionic-app-scripts](https://github.com/ionic-team/ionic-app-scripts/blob/master/package.json)
this `"postcss": "^6.0.13"` installs also the new patch `6.0.15`, which has the new warning.

**Comment by thezakman** [comment]
was this fixed?

**Comment by StefanRein** [comment]
@thezakman Not yet. The issue is open. The warning still exists.

**Comment by AlbenJohn** [comment]
find sass.js file. under /node_modules/@ionic/app-scripts/dist/sass.js.
find var postcssOptions and add from: undefined,

Finally should be like this:
var postcssOptions = {
            from: undefined,
            to: path_1.basename(sassConfig.outFile),
            map: autoPrefixerMapOptions
        };

**Comment by StefanRein** [comment]
https://github.com/ionic-team/ionic-app-scripts/pull/1364#issuecomment-358557878

**Comment by hanifmazlan** [comment]
i solve problem using this ( thanx also to @AlbenJohn)
[https://stackoverflow.com/a/48315414](url)

**Comment by polfernandez** [comment]
Any fix on this?

**Comment by johnroper100** [comment]
While it is not breaking my builds, this warning appears for me as well. This is a one line quick fix, is there already a pull request for this?

**Comment by StefanRein** [comment]
@johnroper100 Yes and it was accepted (it's linked above). But someone should look if the map files are wrong or just in some cases wrong. I remember that in my projects the source files were wrong. I could not see which sass file was the source in the debugger.

**Comment by JosepAlacid** [comment]
This warning still apears.

**Comment by Veyhunk** [comment]
This warning still apears.

**Comment by ajmalhassan** [comment]
This warning still appears.

**Comment by bobbino** [comment]
True, it does, but as @StefanRein suggests, it's still miles away from any production-like environment. better to avoid node_modules. Still waiting for a proper fix. Any updates from the Ionic Team?

**Comment by kirillgroshkov** [comment]
This warning still apears.

**Comment by ElissandroMendes** [comment]
This warning still apears.

**Comment by thorizer** [comment]
This warning still apears.

**Comment by erikcunha** [comment]
This warning still apears.

**Comment by alexgarrett** [comment]
THIs WaRniNg StiLl apPeARs

**Comment by xuan311** [comment]
This warning still apears.

**Comment by StefanRein** [comment]
Please stop. This is not helping in any way. 
This is an open source project. If the people have time, they will fix it. 
There are solutions mentioned above to fix the warning.

**Comment by jhonnhenry** [comment]
This warning still apears.

**Comment by emcniece** [comment]
@jhonnhenry dude, seriously? We know - the issue is still open. Just subscribe and be on your way...

**Comment by mcendon** [comment]
This warning still apears.

**Comment by trsh** [comment]
Still got his warning. Jumping into node_modules is no serious ::/
